internal/proxy/proxy.go

```diff
@@ -213,21 +216,21 @@ type Client struct {
 	// connCount. If not set, there is no limit.
 	maxConns uint64
 
-	cmd    *cobra.Command
 	dialer cloudsql.Dialer
 
 	// mnts is a list of all mounted sockets for this client
 	mnts []*socketMount
+
+	logger cloudsql.Logger
 }
 
 // NewClient completes the initial setup required to get the proxy to a "steady" state.
-func NewClient(ctx context.Context, cmd *cobra.Command, conf *Config) (*Client, error) {
+func NewClient(ctx context.Context, d cloudsql.Dialer, l cloudsql.Logger, conf *Config) (*Client, error) {
```

Comment:
nit: I'm still at of a bit of a loss as to why this is preferable to just passing the cmd in